src/formatter.rs

```diff
@@ -353,10 +353,26 @@ fn color_for_name(color_name: &str) -> Option<Color> {
 
 impl<W: Write> Write for ColorFormatter<W> {
     fn write(&mut self, data: &[u8]) -> Result<usize, Error> {
-        if self.has_pending_style_change {
-            self.write_new_style()?;
+        for line in data.split_inclusive(|b| *b == b'\n') {
```

Comment:
Maybe better to copy the commit message as inline comment? It'll help understand why we have to split lines and write one by one.

Reply:
Good idea. Will do.

Reply:
Oops, I forgot to actually do that. Working on it.